Description: Make sure that all scheduled jobs are finished before
 running a new test. Fixes failures on slow architectures like armel and
 mipsel.
Bug-Debian: http://bugs.debian.org/670878
Last-Update: 2013-02-18

--- a/test/run-stp-tests.sh
+++ b/test/run-stp-tests.sh
@@ -666,6 +666,11 @@
 echo "<PRE>" >>$strfile
 
 for file in 5*.sh; do
+	#
+	# Make sure the past jobs are done before going on.
+	#
+	./waitjobs.sh 1800
+
 	echo $ac_n "Performing $file: $ac_c"
 	echo "" >>$strfile
 	echo "\"$file\":" >>$strfile
